src/state/reducers/search.js

```diff
@@ -9,25 +9,39 @@ import ActionTypes from '../actions/action-types';
 export const searchesReducer = (state = {}, action) => {
   switch (action.type) {
     case ActionTypes.REQUEST_SEARCH:
+      if (((state[action.windowId] || {})[action.companionWindowId] || {}) !== action.query) {
```

Comment:
What would you think about pulling `(state[action.windowId] || {})[action.companionWindowId] || {}` this up into a named constant (since it appears to be in at least 8 places in this file).

Not considering this a blocking comment, it was already duplicated a bit, but it might benefit from giving it a name to help follow along w/ the code.

Reply:
Done.